bugbug/model_calibration.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from sklearn.isotonic import IsotonicRegression
+from sklearn.metrics import mean_squared_error
+from sklearn.model_selection import train_test_split
+
+
+class IsotonicRegressionCalibrator:
+    def __init__(self, model):
+        self.model = model
+        self.calibrator = IsotonicRegression()
+
+    def split_data(self, X, y):
+        X_train_val, X_test, y_train_val, y_test = train_test_split(
+            X, y, test_size=0.2, random_state=42
+        )
+        X_train, X_val, y_train, y_val = train_test_split(
+            X_train_val, y_train_val, test_size=0.2, random_state=42
+        )
+        return X_train, X_val, X_test, y_train, y_val, y_test
+
+    def fit(self, X_train, y_train, X_val, y_val):
```

Comment:
> @suhaibmujahid so im trying to train the regressor model to compare the performance with and without the calibration, but running the training script gives me the following error -
"TypeError: fit() missing 2 required positional arguments: 'X_val' and 'y_val'"
it traces the error back to model.py at the 374th line i.e.
scores = cross_validate(
pipeline, X_train, self.le.transform(y_train), scoring=scorings, cv=5
)
i tried debugging it but couldnt find a soln, do you have any idea?

The signature of the `fit()` method here is not matching the scikit-learn API.

Reply:
which fit() method? the one in the IsotonicRegressionCalibrator class?

Reply:
Also i have to make two more methods predict_proba() and predict() only right?

Reply:
Also how to make the api for the class? just having  those three methods you mentioned fit(), predict_proba() and predict() will do the work right?

Reply:
> but the fit() method of the IsotonicRegressionCalibrator class has positional arguments 'X_val' and 'y_val'" , so why its showing "TypeError: fit() missing 2 required positional arguments: 'X_val' and 'y_val'"?

This is the problem, the `fit()` method on `IsotonicRegressionCalibrator` is not compatible, it asks for 'X_val' and 'y_val' where a scikit-learn compatible API should not.


> Also how to make the api for the class? just having those three methods you mentioned fit(), predict_proba() and predict() will do the work right?

The public methods on a class are the API. You should try and investigate more yourself.